js/sourceadapters.js

```diff
@@ -1701,7 +1703,52 @@ BAMRangerFeatureSource.prototype = Object.create(BAMFeatureSource.prototype);
 BAMRangerFeatureSource.prototype.init = function () {
     this.bamHolder = new Awaited();
     this.bamHolder.provide({}); //TODO because I delay the bam loading
-}
+};
+
+BAMRangerFeatureSource.prototype._featuresFromCache = function (chr, regionStart, regionEnd) {
```

Comment:
Here and in other functions that take chr, start, end attrs, to safeguard from programming errors, check that these attr are defined and not null.